docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,375 @@
+# Formatted ORAS CLI output
+
+## Table of Contents
+
+- [Background](#background)
+- [Guidance](#guidance)
+- [Scenarios](#scenarios)
+  - [Scripting](#scripting)
+  - [CI/CD](#cicd)
+- [Proposal and desired user experience](#proposal-and-desired-user-experience)
+  - [oras manifest fetch](#oras-pull)
+  - [oras pull](#oras-pull)
+  - [oras push](#oras-push)
+  - [oras attach](#oras-attach)
+  - [oras discover](#oras-discover)
+- [FAQ](#faq)
+
+## Background
+
+ORAS has prettified output designed for humans. However, for machine processing, especially in automation scenarios, like scripting and CI/CD pipelines, developers want to perform batch operations and chain different commands with ORAS, as well as filtering, modifying, and sorting objects based on the outputs that are emitted by the ORAS command. Developers expect that ORAS output can be emitted as machine-readable text instead of only the prettified or tabular data, so that it can be used to perform other operations.
+
+The formatted output is not intended to supersede the prettified human-readable and friendly output text of ORAS CLI. It aims to provide multiple views and a programming-friendly experience for developers, DevOps engineers, IT professionals who want to automate their workflows without needing to parse unstructured text.
+
+## Guidance
+
+ORAS allows to use `--output` to output a file or directory in the filesystem. To enable users to format the metadata output of ORAS commands and compute figures based on the formatted output, these two options are proposed as follows:
+
+- Use `--format json|tree|table|go-template='{{ GO_TEMPLATE }}'` to format metadata output of ORAS commands into different formats including prettified JSON, tree, table view, and Go template. It supports computing figures within the template using [Sprig](http://masterminds.github.io/sprig/) functions. Make `-f` as an alias of `--format`.
```

Comment:
Suggest to avoid `'{{` and `}}'` since go template doesn't necessarily need to be surrounded by those.

```suggestion
- Use `--format json|tree|table|go-template=GO_TEMPLATE` to format metadata output of ORAS commands into different formats including prettified JSON, tree, table view, and Go template. It supports computing figures within the template using [Sprig](http://masterminds.github.io/sprig/) functions. Make `-f` as an alias of `--format`.
```


How docker docs lopks like
```console
'TEMPLATE':         Print output using the given Go template.
```
How kubectl doc looks like:
```console
[[-o | --output]=OUTPUT_FORMAT]

```

Reply:
> Make `-f` as an alias of `--format`.

`-f` cannot be used as a shortcut since 
1) instinctively it mean `--force`
2) we already use it as `--force` in `manifest/blob delete` commands

Reply:
Doesn't `go-template='{{ GO_TEMPLATE }}'` makes it more consistent with other tools, and also more familiar for people who use other tools?

Reply:
@toddysm I am suggesting `--format go-template=GO_TEMPLATE` in this doc. Go template string doesn't necessarily need to be surrounded by `{{` and `}}`. Docker uses `TEMPLATE` and kubectl uses `-o=go-template=<template>`

Reply:
- Removed the `-f` alias
- Changed `--format go-template='{{ GO_TEMPLATE }}'` to `--format go-template= GO_TEMPLATE` after checking https://pkg.go.dev/text/template